### PR TITLE
searching_in_sorted_rotated_array.cpp

### PR DESCRIPTION
**Pull request by rohitsigar**
File content  -
1. optimal code in c++
2. valuable comments
3. Time complexity and Space Complexity Analysis
4. Test Cases with explanation
5. Complete explanation

### CODE REVIEW

File: Code/C++/searching_in_sorted_rotated_array.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+i.e - array - [1,3,5,7,8,9] is a sorted array let pivot = 3(this is the index and here zero-based indexing is used)
+                                                                index - [0,1,2,3,4,5]
+around which the array is rotated then resultant array would be array - [8,9,1,3,5,7]
+
+Now this algorithm is used to find the index of key element(which we need to find) in rotated array in O(logn).
+
+key Observation of Algorithm =>
+ -> if we choose any index and split the array into two then one of the two sub array is sorted.
+
+Test case 1 :
```
Comment (plazzy99) [inline Code/C++/searching_in_sorted_rotated_array.cpp:13]: take these test cases to the bottom of the code above the time and space complexity

File: Code/Java/searching_in_sorted_rotated_array.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                                                                index - [0,1,2,3,4,5]
+around which the array is rotated then resultant array would be array - [8,9,1,3,5,7]
+
+Now this algorithm is used to find the index of key element(which we need to find) in rotated array in O(logn).
+
+key Observation of Algorithm =>
+ -> if we choose any index and split the array into two then one of the two sub array is sorted.
+
+Test case 1 :
+  input -
```
Comment (plazzy99) [inline Code/Java/searching_in_sorted_rotated_array.java:14]: same here

**Review by plazzy99 — CHANGES_REQUESTED** [review]
Update the readme for both the code

**Comment by plazzy99** [comment]
One important thing you have to raise another pr with java code and its readme. and update this pr with only cpp code

As we accept only 1 pr with 1 language linked with 1 issue

**Comment by rohitsigar** [comment]
@plazzy99 I have shifted the Position of Test cases in the C++ file and once this PR is done I will do the same  for the java file

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=827846 to learn about workspace recommendations.
```
Comment (plazzy99) [inline .vscode/extensions.json:2]: delete this file

**Review by plazzy99 — CHANGES_REQUESTED** [review]

File: .vscode/extensions.json (hunk 3 above)
Comment (rohitsigar) [reply .vscode/extensions.json]: I have removed .vscode/extensions.json  !!

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
link it to the folder.

**Comment by rohitsigar** [comment]
I have updated the readme file please review it.

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
remove the cpp code

**Comment by rohitsigar** [comment]
I have removed the CPP file.

**Pull request merged**